lib/elven_gard/helpers/basic_encoder.ex

```diff
@@ -38,13 +43,13 @@ defmodule ElvenGard.Helpers.PacketEncoder do
 
   @doc """
   If not already done by the `decode` function, this function will transform his
-  result into a list.
-  This function must return a list starting with your packet header followed by params.
+  result into a tuple.
+  This function must return a tuple starting with your packet header followed by params.
   """
-  @callback post_decode(data :: term, client :: Client.t()) :: list
+  @callback post_decode(data :: term, client :: Client.t()) :: {term, map} | list(tuple)
```

Comment:
Having multiple return types is subject to errors and necessitates adding runtime type checking.

Reply:
This function can returns a packet (`{packet_header, params}`) or a packet list (`[{packet_header, params}]`)

Reply:
I mean that a function should do one thing and not decide how to return something based on the execution condition for example `Foo.get' should return a struct and `Foo.all' should return a container that contains struct.

If for some reason the function `Foo.get` have duplicate key we need runtime match such as:

```elixir
case Foo.get(:bar) do
  [%{}] -> ...
  %{} -> ...
  nil -> ...
end
```

Reply:
Considering that the return of the function is managed by Elven and not by the user, I don't think that this is really a problem : https://github.com/ImNotAVirus/ElvenGard_V2/blob/feature-binary_packets/lib/elven_gard/helpers/frontend.ex#L159

Reply:
Seems okay, but still. I think it's a bad habit to build polymorphous return type on functions except for pattern match such as `gen_statem` (`:keep_state_and_data`, `{:keep_state, data}`...).